src/content/en/showcase/2016/service-worker-perf.md

```diff
@@ -0,0 +1,679 @@
+    // results by individual metric values at reporting time.
+    fields[customDimensions.METRIC_VALUE] = String(fields.eventValue);</strong>
+
+    ga('send', 'event', fields);
+  }
+}
+</pre>
+
+The Google Analytics web interface does not currently provide a way to visualize the distribution of arbitrary metric values, but with the help of the [Google Analytics Core Reporting API](/analytics/devguides/reporting/core/v4/) and the [Google Charts library](/chart/interactive/docs/gallery/histogram) we could query for the raw results and then construct a histogram ourselves.
+
+For example, the following API request configuration was used to get a distribution of `firstpaint` values on desktop with a non-controlled service worker.
+
+<pre class="prettyprint">
+{
+  dateRanges: [{startDate: '2016-05-16', endDate: '2016-05-22'}],
+  metrics: [{expression: 'ga:totalEvents'}],
+  dimensions: [{name: 'ga:dimension2'}],
+  dimensionFilterClauses: [
+    {
+      operator: 'AND',
+      filters: [
+        {
+          dimensionName: 'ga:eventAction',
+          operator: 'EXACT',
+          expressions: ['firstpaint']
+        },
+        {
+          dimensionName: 'ga:dimension1',
+          operator: 'EXACT',
+          expressions: ['supported']
+        },
+        {
+          dimensionName: 'ga:deviceCategory',
+          operator: 'EXACT',
+          expressions: ['desktop']
+        }
+      ],
+    }
+  ],
+  orderBys: [
+    {
+      fieldName: 'ga:dimension2',
+      orderType: 'DIMENSION_AS_INTEGER'
+    }
+  ]
+}
+</pre>
+
+This API request returns an array of values that look like this (Note: these are just the first five results). The results are sorted from smallest to largest, so these rows represent the fastest times.
+
+<table>
+  <tr><th colspan="2">API Response Results (first five rows)</th></tr>
+  <tr class="alt">
+    <td><b>ga:dimension2</b></td>
+    <td align="right"><b>ga:totalEvents</b></td>
+  </tr>
+  <tr>
+    <td>4</td>
+    <td align="right">3</td>
+  </tr>
+  <tr>
+    <td>5</td>
+    <td align="right">2</td>
+  </tr>
+  <tr>
+    <td>6</td>
+    <td align="right">10</td>
+  </tr>
+  <tr>
+    <td>7</td>
+    <td align="right">8</td>
+  </tr>
+  <tr>
+    <td>8</td>
+    <td align="right">10</td>
+  </tr>
+</table>
+
+Here's what these results mean in plain English:
+
+* There were 3 events where the `firstpaint` value was 4 ms
+* There were 2 events where the `firstpaint` value was 5 ms
+* There were 10 events where the `firstpaint` value was 6 ms
+* There were 8 events where the `firstpaint` value was 7 ms
+* There were 10 events where the `firstpaint` `value` was 8 ms
+* etc.
+
+From these results we can extrapolate the `firstpaint` value for every single event and create a histogram of the distribution. We did this for each of the queries we ran.
+
+Here's what the distribution looked like on desktop with a non-controlled (but supported) service worker:
```

Comment:
The service worker status `!=` "controlled". I see why you're asking this, but this way of stating the status is used throughout the case study. I'm not sure it makes sense to change it everywhere. WDYT?